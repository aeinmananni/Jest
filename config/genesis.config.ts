import { BlockType } from "../models";
import { block } from "../script/block";

const INITAIL_DIFFICULTY = 3;
export const MINE_RATE = 1000 //برابر با هزار میلی ثانیه و در واقع میخواهیم هر بیک ثانیه یک بار یک بلاک جدید به شبکه اضافه شود

export const GENESIS: BlockType<string> = block({
  lastHash: "Gen-lastHash",
  hash: "Gen-hash",
  difficulty: INITAIL_DIFFICULTY,
  nonce: 0,
  data: "Gen-data",
});
