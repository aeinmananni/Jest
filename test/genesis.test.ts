import { GENESIS } from '../config/genesis.config';

describe('Genesis', () => {
  //1-  نمونه ای از کلاس ابجکت باشد GENESIS  انتظار دارم ابجکت
  it('returns Genesis Objects', () => {
    expect(GENESIS instanceof Object).toEqual(true);
  });

  // 1- تست اینکه مقدارش دقیقا برابر چیزی باشه که تعریف کردی
  it('hash corretct properties in GENESIS object', () => {
    expect(GENESIS.lastHash).toBe('Gen-lastHash');
    expect(GENESIS.hash).toBe('Gen-hash');
    expect(GENESIS.data).toBe('Gen-data');
  });

  //2- تست اینکه کلیدهای ضروری داخلش باشن (ساختارش تغییر نکنه)
  it('GENESIS should contain lastHash,hash,data', () => {
    expect(GENESIS).toHaveProperty('timeStamp');
    expect(GENESIS).toHaveProperty('lastHash');
    expect(GENESIS).toHaveProperty('hash');
    expect(GENESIS).toHaveProperty('data');
  });

  //3- تست اینکه فقط همون کلیدها رو داشته باشه (بیشتر/کمتر نشه)
  it('GENESIS should not have extra properties', () => {
    const keys = Object.keys(GENESIS);
    expect(keys).toEqual(['lastHash', 'hash', 'data', 'timeStamp']);
  });

  // 4- تست مقدارتایپ ها
  it('GENESIS values should be string type', () => {
    expect(typeof GENESIS['lastHash']).toBe('string');
    expect(typeof GENESIS['hash']).toBe('string');
    expect(typeof GENESIS['data']).toBe('string');
    expect(typeof GENESIS['timeStamp']).toBe('number');
  });

  // 5- برای تست تغییررات ناخواسته
  it('matches the GENESIS snapshot', () => {
    const { timeStamp, ...rest } = GENESIS;
    expect(rest).toMatchSnapshot();
  });
});
