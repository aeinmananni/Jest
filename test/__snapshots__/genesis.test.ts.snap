// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Genesis matches the GENESIS snapshot 1`] = `
{
  "data": "Gen-data",
  "hash": "Gen-hash",
  "lastHash": "Gen-lastHash",
}
`;
